src/core.c/CompUnit/Repository/NQP.pm6

```diff
@@ -1,38 +1,36 @@
 class CompUnit::Repository::NQP does CompUnit::Repository {
-    method need(
-        CompUnit::DependencySpecification $spec,
-        CompUnit::PrecompilationRepository $precomp = self.precomp-repository(),
-        --> CompUnit:D)
-    {
-        if $spec.from eq 'NQP' {
-            my $nqp := nqp::gethllsym('Raku', 'ModuleLoader');
+    my constant %opts = :from<NQP>,;  # comma needed to make a Map
 
-            return CompUnit.new(
-                :short-name($spec.short-name),
-                :handle(CompUnit::Handle.new($nqp.load_module($spec.short-name, {:from<NQP>}))),
-                :repo(self),
-                :repo-id($spec.short-name),
-                :from($spec.from),
-            );
+    method need(CompUnit::Repository::NQP:D:
+      CompUnit::DependencySpecification $spec,
+      CompUnit::PrecompilationRepository $precomp?,
+    --> CompUnit:D) {
+        if $spec.from eq 'NQP' {
+            my $key := $spec.short-name;
+            CompUnit.new:
+              :short-name($key),
+              :handle(CompUnit::Handle.new(
+                 nqp::gethllsym('Raku','ModuleLoader').load_module($key, %opts)
+              )),
+              :repo(self),
+              :repo-id($key),
+              :from<NQP>;
+        }
+        elsif self.next-repo -> $repo {
+            $repo.need($spec, self.precomp-repository)
```

Comment:
This doesn’t use $precomp like it should

Reply:
Good catch.  The idea was to not have it need to set up the precomp repos if not needed.  But *if* they're passed, they should be passed  :-)   Fixed with https://github.com/rakudo/rakudo/commit/8a1661bc2f